presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ImplementIntersectAndExceptAsUnion.java

```diff
@@ -195,22 +242,33 @@ private AggregationNode computeCounts(UnionNode sourceNode, List<Symbol> origina
                     originalColumns,
                     aggregations.build(),
                     signatures.build(),
-                    ImmutableMap.<Symbol, Symbol>of(),
+                    ImmutableMap.of(),
                     ImmutableList.of(originalColumns),
                     Step.SINGLE,
                     Optional.empty(),
                     1.0,
                     Optional.empty());
         }
 
-        private FilterNode addFilter(AggregationNode aggregation)
+        private FilterNode addFilterForIntersect(AggregationNode aggregation)
         {
             ImmutableList<Expression> predicates = aggregation.getAggregations().keySet().stream()
                     .map(column -> new ComparisonExpression(GREATER_THAN_OR_EQUAL, column.toSymbolReference(), new GenericLiteral("BIGINT", "1")))
                     .collect(toImmutableList());
             return new FilterNode(idAllocator.getNextId(), aggregation, ExpressionUtils.and(predicates));
         }
 
+        private FilterNode addFilterForExcept(AggregationNode aggregation, List<Symbol> symbols)
+        {
+            ImmutableList.Builder<Expression> predicatesBuilder = ImmutableList.builder();
+            predicatesBuilder.add(new ComparisonExpression(GREATER_THAN_OR_EQUAL, symbols.get(0).toSymbolReference(), new GenericLiteral("BIGINT", "1")));
```

Comment:
It's a little fragile to treat the first element for the list in a different way from everything else. Since they are materially different, it might be cleaner to separate them explicitly in the argument list.